engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/meta/GetConstructorFieldNamesNode.java

```diff
@@ -13,12 +13,16 @@
     description = "Gets the field names of a constructor.",
     autoRegister = false)
 public class GetConstructorFieldNamesNode extends Node {
-  Array execute(AtomConstructor atom_constructor) {
-    ArgumentDefinition[] fields = atom_constructor.getFields();
-    Object[] result = new Object[fields.length];
-    for (int i = 0; i < fields.length; i++) {
-      result[i] = Text.create(fields[i].getName());
+  Array execute(Object obj) {
+    if (obj instanceof AtomConstructor atom_constructor) {
```

Comment:
Shouldn't we specialize instead? Seem more Truffle-friendly, I think

Reply:
Redone in [0a6ac0a](https://github.com/enso-org/enso/pull/5869/commits/0a6ac0ae980312bcc3ce69194673f548bd3a8840)